cmd/kubeadm/app/cmd/config.go

```diff
@@ -571,11 +584,43 @@ type ImagesList struct {
 	cfg *kubeadmapi.InitConfiguration
 }
 
+// imageTextPrinter prints image info in a text form
+type imageTextPrinter struct {
+	output.TextPrinter
+}
+
+// PrintObj is an implementation of ResourcePrinter.PrintObj for plain text output
+func (itp *imageTextPrinter) PrintObj(obj runtime.Object, writer io.Writer) error {
+	for _, img := range obj.(*outputapiv1alpha1.Images).Images {
+		if _, err := fmt.Fprintln(writer, img.Name); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// imageTextPrintFlags provides flags necessary for printing image in a text form.
+type imageTextPrintFlags struct{}
+
+// ToPrinter returns kubeadm printer for the text output format
+func (ipf *imageTextPrintFlags) ToPrinter(outputFormat string) (output.Printer, error) {
+	if outputFormat == output.TextOutput {
+		return &imageTextPrinter{}, nil
+	}
+	return nil, genericclioptions.NoCompatiblePrinterError{OutputFormat: &outputFormat, AllowedFormats: []string{output.TextOutput}}
+}
+
 // Run runs the images command and writes the result to the io.Writer passed in
-func (i *ImagesList) Run(out io.Writer) error {
+func (i *ImagesList) Run(out io.Writer, printer output.Printer) error {
 	imgs := images.GetControlPlaneImages(&i.cfg.ClusterConfiguration)
+
+	var outImages []outputapiv1alpha1.Image
```

Comment:
Ideally, we want to fill in stuff into an internal type. Then we can convert that to whichever version we need prior to dumping.

Reply:
now it's a slice of strings as you've suggested.